x/programs/cmd/simulator/cmd/config.go

```diff
@@ -54,17 +53,17 @@ func newResponse(id int) *Response {
 
 type Response struct {
 	// The index of the step that generated this response.
-	ID int `json:"id" yaml:"id"`
+	ID int `json:"id"`
 	// The result of the step.
-	Result *Result `json:"result,omitempty" yaml:"result,omitempty"`
+	Result *Result `json:"result,omitempty"`
 	// The error message if available.
-	Error string `json:"error,omitempty" yaml:"error,omitempty"`
+	Error string `json:"error,omitempty"`
 }
 
 func (r *Response) Print() error {
 	jsonBytes, err := json.Marshal(r)
 	if err != nil {
-		return fmt.Errorf("failed to marshal response: %s", err)
+		return fmt.Errorf("failed to marshal response: %w", err)
```

Comment:
It's adding new capabilities to the returned error so it can be manipulated by the callers, it seems that there is just no downsides: https://go.dev/wiki/ErrorValueFAQ